src/dialect/redshift.rs

```diff
@@ -41,10 +41,24 @@ impl Dialect for RedshiftSqlDialect {
     /// treating them as json path. If there is identifier then we assume
     /// there is no json path.
     fn is_proper_identifier_inside_quotes(&self, mut chars: Peekable<Chars<'_>>) -> bool {
+        // PartiQL (used as json path query language in Redshift) uses square bracket as
+        // a start character and a quote is a beginning of quoted identifier.
+        // Skipping analyzing token such as `"a"` and analyze only token that
+        // can be part of json path potentially.
+        // For ex., `[0]`, `['a']` (seems part of json path) or `["a"]` (normal quoted identifier)
+        if let Some(quote_start) = chars.peek() {
+            if *quote_start == '"' {
+                return true;
+            }
+        };
```

Comment:
On `main`, an expression like `SELECT foo["bar"]` evaluates to a `JsonAccess` with the path containing an ident named `bar` but I guess that's wrong. What's a bit interesting though is that with this change, it will evaluate to an expression with alias (`SELECT foo AS ["bar"]`). I know this library doesn't care about what's actually valid SQL, but I wonder if it's an unwanted side effect? I guess it just drops the `[]` and parses `SELECT foo "bar"` which is short for `SELECT foo AS "bar"`.

Another thing, is it intended to end up with `[]` as the quoting style and the value containing the quotes (`"a"`)?  Right now `["a"]` evaluates to an ident which I guess is what's causing above. Is `"a"` different from `a` as an identifier?

```rust
Ident {
    value: "\"1\"",
    quote_style: Some(
        '[',
    ),
},
```

If we encounter a `"` as the first character in `[]` quotes, should we somehow verify what's quoted or ensure we have a closing `"`? If we don't, anything will go, e.g. `["=[]` would parse to an `Ident` with value `"[=` but maybe that's valid PartiQL?

Reply:
Thank you for feedback!

A goal of this MR is improving implementation of RedShift dialect. The method `is_proper_identifier_inside_quotes`  is implemented only for RedShift and I assume that it added only for RedShift specific case.

I verified one more time and found that `SELECT foo["bar"] FROM table_name` or `SELECT foo['bar'] FROM table_name` is not valid in RedShift case. `SELECT foo."bar" FROM table_name` is only valid for this certain case. We don't need to care about this case. I updated comments.

I'm going to improve verification of quoted identifier in this special case that you mentioned. Thank you for highlighting this edge case!

Reply:
I added more unit-tests and updated a code of tokeniser to handle mentioned edge cases.